File: support_tools/swigwin-3.0.12/Examples/test-suite/python/compactdefaultargs_runme.py
from compactdefaultargs import *

defaults1 = Defaults1(1000)
defaults1 = Defaults1()

if defaults1.ret(10.0) != 10.0:
    raise RuntimeError

if defaults1.ret() != -1.0:
    raise RuntimeError

defaults2 = Defaults2(1000)
defaults2 = Defaults2()

if defaults2.ret(10.0) != 10.0:
    raise RuntimeError

if defaults2.ret() != -1.0:
    raise RuntimeError

if defaults2.nodefault(-2) != -2:
    raise RuntimeError

File: support_tools/swigwin-3.0.12/Examples/test-suite/go/compactdefaultargs_runme.go
package main

import . "./compactdefaultargs"

func main() {
	defaults1 := NewDefaults1(1000)
	defaults1 = NewDefaults1()

	if defaults1.Ret(10.0) != 10.0 {
		println(1, defaults1.Ret(10.0))
		panic(defaults1.Ret(10.0))
	}

	if defaults1.Ret() != -1.0 {
		println(2, defaults1.Ret())
		panic(defaults1.Ret())
	}

	defaults2 := NewDefaults2(1000)
	defaults2 = NewDefaults2()

	if defaults2.Ret(10.0) != 10.0 {
		panic(defaults2.Ret(10.0))
	}

	if defaults2.Ret() != -1.0 {
		panic(defaults2.Ret())
	}
}
